internal/datastore/mysql/migrations/driver.go

```diff
@@ -0,0 +1,91 @@
+package migrations
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/authzed/spicedb/internal/datastore/common"
+
+	sqlDriver "github.com/go-sql-driver/mysql"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	errUnableToInstantiate       = "unable to instantiate MysqlDriver: %w"
+	mysqlMissingTableErrorNumber = 1146
+)
+
+type MysqlDriver struct {
+	db *sql.DB
+}
+
+// https://dev.mysql.com/doc/refman/8.0/en/connecting-using-uri-or-key-value-pairs.html
+// URI: [scheme://][user[:[password]]@]host[:port][/schema][?attribute1=value1&attribute2=value2...
+
+/* scheme: The transport protocol to use. Use mysqlx for X Protocol connections and mysql for classic
+   MySQL protocol connections. If no protocol is specified, the server attempts to guess the protocol.
+   Connectors that support DNS SRV can use the mysqlx+srv scheme (see Connections Using DNS SRV Records). */
+// schema: The default database for the connection. If no database is specified, the connection has no default database.
+
+// NewMysqlDriver creates a new driver with active connections to the database specified.
+func NewMysqlDriver(url string) (*MysqlDriver, error) {
+	// TODO: we're currently using a DSN here, not a URI
+	dbConfig, err := sqlDriver.ParseDSN(url)
```

Comment:
need to double check, but I believe the postgres and crdb drivers take a DSN or a URI, not one or the other.

Reply:
Yes, I see that for `crdb` and `postgres`.  Unfortunately, it looks like the Go sql driver [only supports DSN](https://pkg.go.dev/github.com/go-sql-driver/mysql#MySQLDriver.Open) for opening connections.  I'm not sure how to resolve this at the moment.

We could certainly document the limitation for the `mysql` datastore/driver.

Reply:
I think it's not the end of the world if we can only support DSN here - spanner, for example, doesn't support anything remotely like a standard db uri, so there will be some variability regardless. But URI connection strings are common for mysql and its variants, so it would be nice if we could support it too.